src/libpsl-native/src/setdate.cpp

```diff
@@ -2,44 +2,36 @@
 //! @author George FLeming <v-geflem@microsoft.com>
 //! @brief set local/system date and time
 
+#include "setdate.h"
+
+#include <assert.h>
 #include <errno.h>
 #include <langinfo.h>
 #include <locale.h>
 #include <string>
 #include <time.h>
 #include <sys/time.h>
-#include "setdate.h"
 
-//! @brief SetDate sets the date and time on local computer.  You must 
-//!      be super-user to set the time.
+//! @brief SetDate sets the date and time on local computer.
+//!      You must be super-user to set the time.
 //!
 //! SetDate
 //!
 //! @param[in] info
 //! @parblock
 //! A struct that contains program to execute and its parameters
 //!
-//! @exception errno Passes these errors via errno to GetLastError:
-//! - ERROR_BAD_ENVIRONMENT: locale is not UTF-8
-//! - ERROR_INVALID_PARAMETER:  time was not passed in correctly
-//! - ERROR_ACCESS_DENIED:  you must be super-user to set the date
-//!
 //! @retval 0 successfully set date
-//! @retval -1 if failure occurred.  To get extended error information, call GetLastError.
+//! @retval -1 if failure occurred.
 //!
-
-int32_t SetDate(const SetDateInfo &info)
+int32_t SetDate(const SetDateInfo info)
 {
     errno = 0;
 
     // Select locale from environment
     setlocale(LC_ALL, "");
     // Check that locale is UTF-8
-    if (nl_langinfo(CODESET) != std::string("UTF-8"))
-    {
-        errno = ERROR_BAD_ENVIRONMENT;
-        return -1;
-    }
+    assert(nl_langinfo(CODESET) == std::string("UTF-8"));
 
```

Comment:
If this assertion is reasonable, a comment explaining why would be good, because my assumption here is that this is something in the environment which the user controls, not our code.